netjsonconfig/backends/openwrt/converters/interfaces.py

```diff
@@ -420,20 +422,30 @@ def __netjson_interface(self, interface):
             interface = method(interface)
         return interface
 
+    def __get_device_config_for_interface(self, interface):
+        device = interface.get('device')
+        name = interface.get('name')
+        if not name and interface.get('proto') in self._dsa_supported_custom_proto:
+            name = interface.get('.name')
+        return self._device_config.get(device, self._device_config.get(name))
+
     def __netjson_dsa_interface(self, interface):
         if self.__is_device_config(interface) or interface.get('bridge_21', None):
             self.__netjson_device(interface)
         else:
-            device = interface.get('device')
-            # adding device property to interface
-            if self._device_config.get(device, None):
-                device_config = self._device_config[device]
+            device_config = self.__get_device_config_for_interface(interface)
+            if device_config:
                 # ifname has been renamed to device in OpenWrt 21.02
                 if device_config.get('type') == 'bridge':
                     interface['ifname'] = 'br-{}'.format(interface.pop('device'))
-                else:
+                elif interface.get('proto') not in self._dsa_supported_custom_proto:
                     interface['ifname'] = interface.pop('device')
-                if device_config.pop('bridge_21', None):
+                if (
+                    device_config.pop('bridge_21', None)
+                    or interface.get('proto') in self._dsa_supported_custom_proto
+                ):
+                    if interface.get('proto') in self._dsa_supported_custom_proto:
+                        del device_config['type']
```

Comment:
@nemesisdesign I need some help to simplify this section of the code.

Reply:
it may not be needed to do this here, you could do add the conditional code in `__get_device_config_for_interface`, can't you?